buildSrc/src/main/groovy/org/elasticsearch/gradle/vagrant/VagrantService.java

```diff
@@ -0,0 +1,199 @@
+/*
+ * Licensed to Elasticsearch under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.elasticsearch.gradle.vagrant;
+
+import org.apache.commons.io.output.TeeOutputStream;
+import org.elasticsearch.gradle.LoggedExec;
+import org.elasticsearch.gradle.LoggingOutputStream;
+import org.elasticsearch.gradle.Util;
+import org.gradle.api.Action;
+import org.gradle.api.Project;
+import org.gradle.internal.logging.progress.ProgressLogger;
+import org.gradle.internal.logging.progress.ProgressLoggerFactory;
+
+import javax.inject.Inject;
+import java.io.File;
+import java.io.OutputStream;
+import java.nio.file.Paths;
+import java.util.Arrays;
+import java.util.Objects;
+import java.util.function.UnaryOperator;
+
+/**
+ * A service to manage a vagrant box.
+ *
+ * This is created alongside a {@link VagrantExtension} for a project to manage starting and
+ * stopping a single vagrant box.
+ */
+public class VagrantService {
+
+    public static class VagrantExecSpec {
```

Comment:
nit: again, my personal preference is for inner classes to be at the end of a class defintion. When I open `VagrantService.java` the first thing I expect to see is code for `VagrantService`. I find having to scroll just to find the constructor and instance variables is confusing.